src/riak_client.erl

```diff
@@ -541,23 +544,23 @@ get_cover_aux(MinPar, ReqId, N, RingSize) ->
                       {error, Reason}),
                     riak_client()) ->
                            list(list({atom(), term()})) | {error, term()}.
-replace_cover(_Bucket, _P, {error, Reason}, _OtherBroken, _Client) ->
+replace_cover(_Mod, _Bucket, _P, {error, Reason}, _OtherBroken, _Client) ->
     {error, Reason};
-replace_cover(Bucket, _P, {ok, Replace}, OtherBroken, _Client) ->
-    pick_cover_replacement(proplists:get_value(subpartition, Replace),
+replace_cover(Mod, Bucket, _P, {ok, Replace}, OtherBroken, _Client) ->
+    pick_cover_replacement(Mod, proplists:get_value(subpartition, Replace),
```

Comment:
Since the 2nd arg to `pick_cover_replacement` is only being used for picking traditional vs subpartition I suggest that the destiction is done here as
````erlang
case proplists:get_value(subpartion, Replace) of
    undefined -> replace_traditional_cover(...);
    _  -> replace_subpartition_cover
end.
````
In my head it makes it easier to understand the branching.